### PR TITLE
Fix fatal issues with Azure deployment templates

### PR DESCRIPTION
**Pull request by glahaye**
### Motivation and Context
Current Azure ARM template is not functional

### Description
- Remove embedding storage methods no longer support (Volatile and PostGres)
- Merge the 3 app insights into 1
- Have a single switch for whether to deploy binary packages
- Remove obsolete MemoryStore app settings
- Add Qdrant settings to SemanticMemory:Services
- Add memory pipeline service to Qdrant vnet

### Contribution Checklist
- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [Contribution Guidelines](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible

### CODE REVIEW

**Review by TaoChenOSU — APPROVED** [review]

**Comment by TaoChenOSU** [comment]
This addresses: https://github.com/microsoft/chat-copilot/issues/510

File: scripts/deploy/deploy-azure.ps1 [hunk 1]
```diff
@@ -80,15 +76,15 @@ param(
     $DebugDeployment,
 
     [switch]
-    # Switches on whether to deploy release packages
-    $NoDeployPackage
+    # Skip deployment of binary packages
+    $DontDeployPackages
```
[the remark this answers is not in the document]
Comment (glahaye) [reply scripts/deploy/deploy-azure.ps1:80]: Yes, I would say this is a small breaking change. Given how it simplifies things and in light of the upcoming simplification of resource deployment, this is something I don't mind.

De we really care about keeping the names between the PowerShell arguments and the Bash ones aligned??

[the remark this answers is not in the document]
Comment (alliscode) [reply scripts/deploy/deploy-azure.ps1:80]: I question if the value gained by renaming this parameter is worth the work people will need to do to update scripts they may be using.

**Review by alliscode — APPROVED** [review]

File: scripts/deploy/deploy-azure.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (glahaye) [reply scripts/deploy/deploy-azure.ps1:80]: Reverting the over-zealous renaming of this argument

**Review by TaoChenOSU — APPROVED** [review]

**Review by dehoward — APPROVED** [review]

**Comment by victor-v-rad** [comment]
@glahaye @TaoChenOSU 
Please fix OpenAI Deployment as well 
https://github.com/microsoft/chat-copilot/issues/516

**Pull request merged**